editors/vscode/esbuild.js

```diff
@@ -66,6 +68,7 @@ esbuild
             "fs",
         ],
         outfile: "out/extension.js",
+        platform: "node",
```

Comment:
do we still need `bundle` and `external`  ?

Reply:
Afaik yes and yes.

The docs say that bundling is never enabled by default.

Similarly, vs code docs say that “require(‘vscode’)” is the only external allowed and a shim is provided: https://code.visualstudio.com/api/extension-guides/web-extensions#web-extension-main-file

Reply:
the docs you pointed out is for web extension, not the normal extension.

Reply:
Serves me well for commenting when sitting on the tram :-). You're right, I mistook this as the browser section (where this would apply).

I do think that you're right, we should not need bundling of anything, and without bundling we shouldn't need externals. However, at the moment we need it AFAICS:

1. `extension.ts` imports variables from `common.ts`. Without bundling that would turn into a `require("./common")` in the generated `out/extension.js`. That can be dealt with by adding another build rule to generate that file.
2. `common.ts` imports types from `../../../tools/online_editor/src/shared/properties`. This is fine, as only types are needed. But once a variable is imported that's needed at run-time, we need to bundle from that import.

This is the caveat of importing from files outside the npm project:

1. Either they're just type imports, then nothing is needed at run-time. We can safeguard this by using `import type { ... }` instead of just `import { ... }`
2. They're imports from another npm package: Then for the node build we can just let that be a dependency, the generated JS code would be `require("somedep")` and via `dependencies` in `package.json` at run-time the module `somedep` will be found under `node_modules` by node (esbuild won't touch).
3. They are imports from another directory: Then it is my believe that we need to bundle those files - unless we also turn them into 2., which includes publishing to the npm registry.

My initial motivation with this patch was to avoid the situation we ran into, where esbuild resolving imports within a module based on the target platform incorrectly to browser, when we want node. I sense from your question that you may want to do more steps in one, so I'll have a go at it.